components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/form-builder/form-utils.js

```diff
@@ -348,59 +768,166 @@ define(['require', 'lodash', 'appData', 'log', 'constants', 'handlebar', 'annota
+                var isPredefined = false;
+                _.forEach(predefinedAnnotations, function (annotation) {
+                    if (savedAnnotation.name.toLowerCase() === annotation.name.toLowerCase()) {
+                        isPredefined = true;
+                        return false;
+                    }
+                });
+                if (!isPredefined) {
+                    userAnnotations.push(savedAnnotation);
                 }
+            });
+            return userAnnotations;
+        };
+
+        /**
+         * @function to create stream attribute object with empty values
+         * @param {Object} streamAttributes
+         * @return {Object} streamAttributesObject
+         */
+        FormUtils.prototype.createStreamAttributesObject = function (streamAttributes) {
+            var streamAttributesObject = [];
+
+            _.forEach(streamAttributes, function (attribute) {
+                streamAttributesObject.push({ key: attribute.getName(), value: "" });
+            })
+            return streamAttributesObject;
+        };
+
+        /**
+         * @function to select the options according to the selected rdbms type
+         * @param {Object} predefined_options all the options of rdbms with the user given values
+         * @return {Object} rdbms_options
+         */
+        FormUtils.prototype.getRdbmsOptions = function (predefined_options) {
+            var rdbms_options = [];
+            var selectedRdbmsType = $('input[name=radioOpt]:checked', '#define-rdbms-type').val();
+            if (selectedRdbmsType == "datasource") {
+                _.forEach(predefined_options, function (predefinedOption) {
+                    if (predefinedOption.name.toLowerCase() === "datasource") {
+                        rdbms_options.push({
+                            name: predefinedOption.name, value: predefinedOption.value, description: predefinedOption
+                                .description, optional: false, defaultValue: predefinedOption.defaultValue
+                        })
+                    } else if (predefinedOption.name.toLowerCase() === "pool.properties" ||
+                        predefinedOption.name.toLowerCase() === "table.name" ||
+                        predefinedOption.name.toLowerCase() === "field.length") {
+                        rdbms_options.push({
+                            name: predefinedOption.name, value: predefinedOption.value, description: predefinedOption
+                                .description, optional: predefinedOption.optional, defaultValue: predefinedOption
+                                    .defaultValue
+                        })
+                    }
+                });
+            } else if (selectedRdbmsType == "inline-config") {
+                _.forEach(predefined_options, function (predefinedOption) {
+                    if (predefinedOption.name.toLowerCase() === "username" ||
```

Comment:
why are we hardcoding the properties ?

Reply:
Fixed in 8dd3e78e1a28eccb6c7014c24a878fd139399e87